tests/ui/async-await/async-fn-size-uninit-locals.rs

```diff
@@ -94,9 +94,9 @@ async fn join_retval() -> Joiner {
 }
 
 fn main() {
-    assert_eq!(2, std::mem::size_of_val(&single()));
+    assert_eq!(3, std::mem::size_of_val(&single()));
     assert_eq!(3, std::mem::size_of_val(&single_with_noop()));
-    assert_eq!(3074, std::mem::size_of_val(&joined()));
+    assert_eq!(3075, std::mem::size_of_val(&joined()));
```

Comment:
How come these futures got bigger?

Reply:
It's due to a problem in the dataflow engine that uses information from unreachable blocks generated from desugaring of drop and replace. In this case the future holds an additional useless drop flag.
Will be fixed independently by https://github.com/rust-lang/rust/pull/106430.